src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java

```diff
@@ -62,37 +65,38 @@ public void testExpiry() throws InterruptedException {
 	}
 	
 	private static class Counter {
-		private int counter;
+		private int ctr;
 		
 		public void increase() {
-			this.counter++;
+			this.ctr++;
 		}
 		public int getCounter() {
-			return this.counter;
+			return this.ctr;
 		}
 	}
-	
+
 	@Test
-	public void testAutoRefresh() throws InterruptedException {
+	public void testAutoRefresh() {
 		final Counter counter = new Counter();
-		
+		String testKey="key";
 		AutoRefreshingCacheMap<String, String> subject = new AutoRefreshingCacheMap<>("test", null, Duration.ofSeconds(10), Duration.ofMillis(500), key -> {
 			counter.increase();
 			return "refreshed";
 		});
-		
-		subject.put("key", "initial");
-		String value = subject.get("key");
-		Assert.assertEquals("initial", value);
+
+		subject.setRefresherThreadWithIncreasedPriority(true);
+		// Note that this also start the refresher thread immediately.
+		// It helps keeping this unit test to stay stable.
+
+		subject.put(testKey, "initial");
+
+		Assert.assertTrue(subject.containsKey(testKey));
+		Assert.assertEquals("initial", subject.get(testKey));
 		
 		// we have to wait until the thread had a chance to refresh
-		Thread.sleep(1000);
-		
-		boolean exists = subject.containsKey("key");
-		Assert.assertTrue(exists);
-		
-		Assert.assertEquals("refreshed", subject.get("key"));
-		Assert.assertEquals(1, counter.getCounter());
+		await().atMost(3, SECONDS).untilAsserted(() -> Assert.assertEquals(1, counter.getCounter()));
```

Comment:
Good idea.
But why so complicated? Wouldn't 

```java
await().atMost(3, SECONDS).until(() -> counter.getCounter() >= 1));
``` 
also work?
At least this would be much easier to understand for me.

Reply:
We were doing an assert on counter being one anyway. So by doing the await for the counter to be 1 , and then the later assert on it.. its kind of wasteful. This gets the assert and the 'condtion' done in one line

Reply:
okay, you convinced me.